cli/src/modules/workflow.rs

```diff
@@ -31,30 +43,82 @@ pub async fn parse_workflow_args(
+                        ..Default::default()
+                    })
+                    .await?;
+                if *sent_to_self {
+                    let invitation_object = invite_url_to_object(connection.invitation_url)?;
+                    agent.receive_invitation(invitation_object).await?;
+                } else {
+                    qr::print_qr_code(&connection.invitation_url)?;
+                    println!("Connection id: {}", connection.connection_id);
+                    println!(
+                        "Use this invitation to connect with your agent.\n{}",
+                        connection.invitation_url
+                    );
+                    copy!("{}", connection.invitation_url);
+                }
+                let limit = timeout.map(|t| t / 1000).unwrap_or(10);
+                debug!("Looping {} times", limit);
+                for i in 1..=limit {
+                    let connection =
+                        ConnectionModule::get_by_id(&agent, connection.connection_id.to_owned())
+                            .await?;
+                    if connection.state != "active" && connection.state != "response" {
+                        trace!("Connection state is not active");
+                        std::thread::sleep(std::time::Duration::from_millis(1000));
+                    } else {
+                        trace!("Connection state is active, continuing the flow");
+                        credential_offer(connection.connection_id, agent).await?;
+                        break;
+                    }
+                    if i == limit {
+                        return Err(Error::InactiveConnection.into());
+                    }
+                }
+            }
+        },
+    };
+    println!("{} executed workflow", "Successfully".green());
+    println!("Depending on the wallet and agent it might take a few seconds for the credential to arrive");
+    loader.stop();
+    Ok(())
+}
+
+async fn credential_offer(
+    connection_id: String,
+    agent: impl ConnectionModule + CredentialsModule + SchemaModule + CredentialDefinitionModule,
+) -> Result<()> {
+    // Mock credential
+    let mut attributes: HashMap<String, String> = HashMap::new();
+    attributes.insert(String::from("Name"), String::from("Joyce Brown"));
+    attributes.insert(String::from("Date Of Birth"), String::from("19890321"));
+    attributes.insert(String::from("Street"), String::from("Main Road 207"));
+    attributes.insert(String::from("City"), String::from("New York"));
+    attributes.insert(String::from("Bank"), String::from("qBank New York"));
+    attributes.insert(
+        String::from("Card Number"),
+        String::from("4537-6696-0666-0146"),
+    );
+    attributes.insert(String::from("Security Code"), String::from("063"));
+    attributes.insert(String::from("Valid Until"), String::from("20251212"));
+    debug!("Mock credential:\n{:#?}", attributes);
+
+    let options = CredentialOfferWorkflow {
```

Comment:
```suggestion
    let workflow = CredentialOfferWorkflow {
```

seems like a clearer name IMO